waspc/src/Analyzer/Parser/Parser.y

```diff
@@ -0,0 +1,119 @@
+{
+-- This file is processed by Happy (https://www.haskell.org/happy/) and generates
+-- the module `Analyzer.Parser.Parser`
+
+module Analyzer.Parser.Parser
+  ( parse
+  ) where
+
+import Analyzer.Parser.Lexer
+import Analyzer.Parser.AST
+import Analyzer.Parser.Token
+import Analyzer.Parser.ParseError
+import Analyzer.Parser.Util (Parser, initialState, ParserState (..))
+import Control.Monad.Trans.State.Lazy (evalStateT, get)
+import Control.Monad.Trans.Except (throwE, runExcept)
+import Control.Monad.Trans.Class (lift)
+}
+
+-- These lines tell Happy to call the main parsing function `parse`, that it
+-- will operate on the `Token` type, and to call `parseError` when the parser
+-- encounters an error.
+%name parse
+%tokentype { Token }
+%error { parseError }
+
+-- This sets up Happy to use a monadic parser and threaded lexer
+-- This means that Happy will request tokens as it needs them instead of
+-- requiring a list of all tokens up front. A monad can also be used in the
+-- lexer and parser to track state and errors.
```

Comment:
```suggestion
-- This sets up Happy to use a monadic parser and threaded lexer.
-- This means that parser generated by Happy will request tokens from lexer as it needs them instead of
-- requiring a list of all tokens up front.
-- Both lexer and parser operate in the 'Parser' monad, which can be used to track shared state and errors.
-- Check https://www.haskell.org/happy/doc/html/sec-monads.html#sec-lexers for more details.
```